Purchases/Purchasing/PurchasesOrchestrator.swift

```diff
@@ -679,12 +684,24 @@ private extension PurchasesOrchestrator {
     }
 
     @available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
-    private func purchase(sk2Product: SK2Product) async -> Result<(CustomerInfo, Bool), Error> {
+    private func purchase(
+        sk2Product: SK2Product,
+        discount: StoreProductDiscountType?
```

Comment:
nit (if we are still doing this): `maybeDiscount`

Reply:
I keep forgetting about this, cause I still don't see the benefit 😕 Swift is a type safe language, so we never add types to variable names (Hungarian Notation). It's like doing things like `productsSet`, `appUserIDString`, `isRestoreBoolean`.
I know it's in the `SwiftStyleGuide`, but it also mentions:
> don't use `maybe` for public properties, since it might be confusing to developers unfamiliar with our naming convention

Having 2 different naming convention for `public` and `internal` APIs feels like a lot of overhead, so I keep forgetting when to use it and when not to.

The guideline for public APIs is consistent with the [Swift API Design Guidelines](https://www.swift.org/documentation/api-design-guidelines/), which says:
> Name variables, parameters, and associated types according to their roles, **rather than their type constraints.**

I'm happy to send an e-mail to make a formal request for this, I know this PR isn't the right place to have this conversation. I would like to make the case for consolidating our naming conventions :)

Reply:
yeah, all very valid points. i'm not a huge fan of it either but i have less swift experience :) i'd just like us to be consistent! i guess the only argument i have _for_ this naming is distinguishing between variables in the case that we do `let discount = maybeDiscount`, but i can get over that and/or maybe there are other ways to accomplish a distinction.

agreed that we don't have to discuss on this PR though, i think a follow-up would be great

Reply:
But you can always do `if let discount = discount`.